module-2/cdk/lib/ECSConstruct.ts

```diff
@@ -0,0 +1,179 @@
+      // Allow the ECS Tasks to download images from ECR
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      // Allow the ECS tasks to upload logs to CloudWatch
+      "logs:CreateLogStream",
+      "logs:CreateLogGroup",
+      "logs:PutLogEvents"
+    );
+    ECSTaskRoleStatement1.addAllResources();
+    ECSTaskRole.addToPolicy(ECSTaskRoleStatement1);
+
+    const ECSTaskRoleStatement2 = new iam.PolicyStatement();
+    ECSTaskRoleStatement2.addActions(
+      //  Allows the ECS tasks to interact with only the MysfitsTable
+      //  in DynamoDB
+      "dynamodb:Scan",
+      "dynamodb:Query",
+      "dynamodb:UpdateItem",
+      "dynamodb:GetItem",
+      "dynamodb:DescribeTable"
+    );
+    ECSTaskRoleStatement2.addResource(
+      "arn:aws:dynamodb:*:*:table/MysfitsTable*"
+    );
+    ECSTaskRole.addToPolicy(ECSTaskRoleStatement2);
+
+    const EcsServiceRole = new iam.Role(this, "EcsServiceRole", {
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal("ecs.amazonaws.com"),
+        new iam.ServicePrincipal("ecs-tasks.amazonaws.com")
+      ),
+      path: "/"
+    });
+    const EcsServiceRoleStatement1 = new iam.PolicyStatement();
+    EcsServiceRoleStatement1.addActions(
+      // Rules which allow ECS to attach network interfaces to instances
+      // on your behalf in order for awsvpc networking mode to work right
+      "ec2:AttachNetworkInterface",
+      "ec2:CreateNetworkInterface",
+      "ec2:CreateNetworkInterfacePermission",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DeleteNetworkInterfacePermission",
+      "ec2:Describe*",
+      "ec2:DetachNetworkInterface",
+
+      // Rules which allow ECS to update load balancers on your behalf
+      //  with the information sabout how to send traffic to your containers
+      "elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
+      "elasticloadbalancing:DeregisterTargets",
+      "elasticloadbalancing:Describe*",
+      "elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+      "elasticloadbalancing:RegisterTargets",
+
+      //  Rules which allow ECS to run tasks that have IAM roles assigned to them.
+      "iam:PassRole",
+
+      //  Rules that let ECS interact with container images.
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+
+      //  Rules that let ECS create and push logs to CloudWatch.
+      "logs:DescribeLogStreams",
+      "logs:CreateLogStream",
+      "logs:CreateLogGroup",
+      "logs:PutLogEvents"
+    );
+    EcsServiceRoleStatement1.addAllResources();
+    EcsServiceRole.addToPolicy(EcsServiceRoleStatement1);
+
+    this.cluster = new ecs.Cluster(this, "Cluster", {
+      clusterName: "MythicalMysfits-Cluster",
+      vpc: core.vpc
+    });
+    this.cluster.connections.allowFromAnyIPv4(new ec2.TcpPort(8080));
```

Comment:
I think `LoadBalancedFargateService` adds the external port by default. @rix0rrr?